src/main/java/com/google/firebase/remoteconfig/Template.java

```diff
@@ -224,4 +273,33 @@ public boolean equals(Object o) {
   public int hashCode() {
     return Objects.hash(etag, parameters, conditions, parameterGroups, version);
   }
+
+  private static class InAppDefaultAdapter implements JsonSerializer<ParameterValue.InAppDefault> {
+
+    @Override
+    public JsonElement serialize(ParameterValue.InAppDefault src, Type typeOfSrc,
+                                 JsonSerializationContext context) {
+      JsonObject obj = new JsonObject();
+      obj.addProperty("useInAppDefault", true);
+      return obj;
+    }
+  }
+
+  private static class VersionAdapter implements JsonSerializer<Version> {
```

Comment:
This wouldn't be needed if we used the existing TemplateResponse and VersionResponse types to handle serialization.